### ISSUE TITLE
AIML tag <THAT> not working 

### ISSUE THREAD

**Issue by ihamzehald** [Program-O/Program-O#110]
Hello sir, i created a fresh bot to test this code "in the link"  where it have "that" to simulate a conversation context but the "that" tag didn't work as expected, the problem is that, whenever the input was the pattern before the "that" tag the parser gonna return that template, based on my understanding to the AIML that tag the parser should only return the category template that have "that" only when the bot response matches the text in "that", i tihnk this is a bug, please advice and let me know if you need more information from my side, this is the code that i tested.
http://www.tutorialspoint.com/aiml/aiml_that_tag.htm

**Comment by Dave-Morton** [comment]
Can you provide a log of a conversation where the chatbot doesn't respond as expected? That would help.

**Comment by ihamzehald** [comment]
Hello @Dave-Morton  this is the chat log based on the AIML in the link :

Seeker: WHAT ABOUT MOVIES
Bella: Do you like comedy movies
Seeker: YES
Bella: Nice, I like comedy movies too.
[the following (YES) shouldn't return that output based on the AIML]
Seeker: YES
Bella: Nice, I like comedy movies too.
Seeker: YES
Bella: Nice, I like comedy movies too.

**Comment by ihamzehald** [comment]
@Dave-Morton kindly note that, i uploaded the same AIML file on pandorabots.com and it worked as expected.

**Comment by Dave-Morton** [comment]
Ah. You're right. That's not the expected behavior. I'll look into it over the weekend and see what I can find.

**Comment by Dave-Morton** [comment]
I had a quick look, and the problem seems to stem from the core search query not including the "thatpattern" keywords in the search. I won't have time to work on the fix until Sunday or Monday, but it's on the ToDo list.

**Comment by ihamzehald** [comment]
Thank you for your time sir, looking forward for the fix, for now i am exploring other alternative (rivescript since they have their own interrupter and support js and php by third party library because i don't have the time to create my own interpreter now) but i am still wanna build a chat bot using AMIL and program-o, looking forward for the fix, regards.

**Comment by Dave-Morton** [comment]
Noah Petherbridge, the creator of Rivescript, is a friend of mine. Rivescript is a good alternative to using AIML, but it's **very** different, and has a bit of a learning curve. :)

We've found and fixed the problem with the THAT tag not being properly (Liz and I knocked it out together, just a few minutes ago), so you might want to grab the latest version and give it a try. A simple "overlay install" (e.g. just copying the new files over) should do the trick.
